src/libAtomVM/nifs.c

```diff
@@ -3553,6 +3560,56 @@ static term nif_atomvm_read_priv(Context *ctx, int argc, term argv[])
     return result;
 }
 
+static term nif_atomvm_set_time_of_day(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+
+    VALIDATE_VALUE(argv[0], term_is_atom);
+    VALIDATE_VALUE(argv[1], term_is_any_integer);
+
+#ifdef HAVE_SETIMEOFDAY
+
+    term unit = argv[0];
+    avm_int64_t value = term_maybe_unbox_int64(argv[1]);
+
+    struct timeval tp;
+    if (unit == SECOND_ATOM) {
+        tp.tv_sec = value;
+        tp.tv_usec = 0L;
+
+    } else if (unit == MILLISECOND_ATOM) {
+        tp.tv_sec = value / 1000;
+        tp.tv_usec = (value % 1000) * 1000;
+
+    } else if (unit == MICROSECOND_ATOM) {
+        tp.tv_sec = value / 1000000;
+        tp.tv_usec = value % 1000000;
+
+    } else {
+        RAISE_ERROR(BADARG_ATOM);
+    }
+
+    struct timezone tz = {
+        .tz_minuteswest = 0,
+        .tz_dsttime = 0
+    };
+    int res = settimeofday(&tp, &tz);
```

Comment:
portability warning: settimeofday is not even a POSIX function, we might investigate how to keep everything as portable as possible.

Reply:
Right, though it’s guarded under a macro.  Putting the Nif in the ESP32 platform could address this, too, I suppose.

Reply:
It does look like `settimeofday` is a POSIX function.

https://manpages.debian.org/buster/manpages-dev/settimeofday.2.en.html

https://docs.oracle.com/cd/E19048-01/chorus4/806-3328/6jcg1bm2u/index.html

Reply:
The POSIX function to set time is `clock_settime`
https://pubs.opengroup.org/onlinepubs/9699919799/

It is available in newlib and is available on ESP-32, at least on 4.4.5.

Reply:
OTH, pico sdk has `settimeofday` and not `clock_settime`.